code/go/0chain.net/smartcontract/storagesc/validator.go

```diff
@@ -193,7 +193,7 @@ func (sc *StorageSmartContract) updateValidator(t *transaction.Transaction,
 			}
 		}
 	}
-	inputValidator.StakePoolSettings = inputValidator.StakePoolSettings
```

Comment:
It should be `savedValidator.StakePoolSettings = inputValidator.StakePoolSettings`, see what was changed 
https://github.com/0chain/0chain/pull/1427/files#diff-a190a605e48ed0a9eb8803f510618ca09c0432844cfa7aeefda1fe62364d13baL155-L158, 

@Sriep please confirm

Reply:
@peterlimg  It should be fine to remove. `saveValidator` is readonly here. new values should be saved by `inputValidator.emitUpdate`?

Reply:
fixed